#165717529 Create code implementation of Interests page from mockup

damafeez opened the pull request:
#### What Does This PR Do?
This PR creates the Interests page

#### Description Of Task To Be Completed
- create route for Interests page to enable navigation
- create code implementation of Interests page from mockup
- allow individual interests to be selectable
- allow selected interests to be deselected
- create action buttons

#### Any Background Context You Want To Provide?
None

#### How can this be manually tested?
- navigate to Interests page and follow steps provided in screenshot

#### What are the relevant github issues?
None

#### Screenshot(s)
![Interests design](https://user-images.githubusercontent.com/26779819/57387042-87045600-71ad-11e9-852c-5e88bb8e2bf3.gif)

olusoladavid commented on client/components/common/DropDown.js:
```diff
@@ -21,12 +21,11 @@ class DropDown extends Component {
       className,
     } = this.props;
-    const classNames = `dropdown ${className && className}`;
+    const classNames = `dropdown ${className}`;
```
Nice insight here @damafeez. Did you consider the possibility that `className` may be undefined here and leads to a classname like `dropdown undefined`?

damafeez replied:
Thank you for your feedback. This has been fixed.

olusoladavid commented on client/components/common/UserProfile.js:
```diff
@@ -11,26 +11,29 @@ const UserProfile = ({
+        <Link to="/interests">My Interests</Link>
+      </li>
+      <li>
+        <Link to="/settings">Settings</Link>
```
This link is not on `mvp-develop` branch. Please check your rebase process again.

damafeez replied:
I did a rebase, the `Interests` link came from @uchemukolo 's PR on `develop`. I should just comment it out?

olusoladavid commented on client/pages/Interests/index.jsx:
```diff
@@ -0,0 +1,141 @@
+ */
+class Interests extends React.Component {
+  state = {
+    interests: [
```
Nice move with the fixtures. I think it will be better if we move it into a separate file as it is just dummy data and not officially a part of the definition of this component.

damafeez replied:
Thank you for your feedback. This has been fixed.

olusoladavid reviewed (commented):
Great job on this implementation @damafeez. I made some comments on how we can make it better. Please check them out. Cheers 🥂

damafeez committed:
feat(interests): create interests page

 - add interest card
 - remove duplicate/bug dropdown
 - add Interests to route
 - make Interests page responsive
 - remove nav button outline

olusoladavid reviewed (approved):
Great job on this implementation @damafeez. Works as required.
